packages/handlersjs-core/lib/daemons/scheduler.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { Scheduler } from './scheduler';
+
+describe('Scheduler', () => {
+
+  let scheduler: Scheduler;
+  const task = jest.fn();
+
+  // checks if the scheduler is running
+  const isRunning = async () => {
+
+    task.mockClear();
+    await new Promise((r) => setTimeout(r, 20)); // sleep 20ms
+    const callAmount = task.mock.calls.length;
+
+    return callAmount >= 3 && callAmount <= 4;
+
+  };
+
+  beforeEach(() => {
```

Comment:
When using mutable variables, put them as close as possible to their mutation (which in tests means: write them just above `beforeEach`/`beforeAll`)
```suggestion
  let scheduler: Scheduler;
  
  beforeEach(() => {
```